### PR TITLE
DS-456: upgrade node version and npm packages

### PR DESCRIPTION
**Pull request by stfurkan**
# READY FOR REVIEW

# Summary
- upgrade node version and npm packages (minor updates)

# Review By (Date)
- End of sprint

# Criticality
- 3

# Review Tasks

## Setup tasks and/or behavior to test

1. Make sure everything works as expected.

### CODE REVIEW

File: src/api/auth/login.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { authInstance } from '../../utilities/authInstance';
 
 const handler = connect()
   // Return user to page they were previously on after logging in.
-  .use((req, res, next) => {
+  .get((req, res, next) => {
```
Comment (stfurkan) [inline src/api/auth/login.js:7]: @sherakama this was causing the `login` issue. Can you please try again? Thanks

File: .github/workflows/integration.yml [hunk 2]
```diff
@@ -40,13 +38,13 @@ jobs:
...
-        uses: cypress-io/github-action@v5
+        uses: cypress-io/github-action@v6
         with:
           # we have already installed all dependencies above
           install: false
           build: npm run build-test
-          start: npm run start-test
-          wait-on: 'http://localhost:8000'
+          start: npm run serve-test
+          wait-on: 'http://localhost:9000'
```
Comment (stfurkan) [inline .github/workflows/integration.yml:47]: @rebeccahongsf redirects were not working as expected in Cypress and some tests were failing. I Madde some changes in Cypress. FYI

File: cypress.config.ts [hunk 3]
```diff
@@ -19,7 +19,7 @@ export default defineConfig({
     runMode: 2,
   },
   e2e: {
-    baseUrl: 'http://localhost:8000',
+    baseUrl: 'http://localhost:9000',
```
Comment (stfurkan) [inline cypress.config.ts:22]: @rebeccahongsf changed port for Cypress test

File: cypress/e2e/2-travel-study/registration-form.cy.ts [hunk 4]
```diff
@@ -19,7 +19,7 @@ describe('Travel-Study Trip Registration Form Page', () => {
     cy.visit('/travel-study/destinations/finland-2022/finland-reg-form/');
 
     // Confirm that the URL matches the expected URL
-    cy.url().should('include', '/travel-study/destinations/finland-2022/finland-reg-form/');
+    cy.url().should('include', '/travel-study/destinations/finland-2022/finland-reg-form');
```
Comment (stfurkan) [inline cypress/e2e/2-travel-study/registration-form.cy.ts:22]: @rebeccahongsf removed `/`

File: package.json [hunk 5]
```diff
@@ -97,43 +96,44 @@
     "cy:run": "CYPRESS_baseUrl=http://localhost:9000 cypress run",
     "test:e2e:ci": "CYPRESS_SUPPORT=y npm run build && start-server-and-test serve http://localhost:9000 cy:run",
     "build-test": "MEGAPROFILE_MOCK=true CYPRESS_SUPPORT=y npm run build",
-    "start-test": "MEGAPROFILE_MOCK=true npm run develop"
+    "start-test": "MEGAPROFILE_MOCK=true npm run develop",
+    "serve-test": "MEGAPROFILE_MOCK=true npm run serve"
```
Comment (sherakama) [inline package.json:100]: Why do we need both?

Comment (stfurkan) [reply package.json:100]: @sherakama  one of them start the development (port 8000), the other starts the port 9000. I thought we might use it for our local tests but I can delete the `start-test` one if you want :)

File: src/api/auth/login.js (hunk 1 above)
Comment (sherakama) [reply src/api/auth/login.js:7]: Ah, no http method. That would do it. 

Can you move this get to line 15 `.use(authInstance.initiate());` instead?

Convention is to have the `.use` statements on the middleware and the HTTP method on the endpoint logic.

File: package.json (hunk 5 above)
Comment (sherakama) [reply package.json:100]: I guess these are good to have as the `start-test` is using the development server and the `serve-test` requires the build. Keep them both.

**Review by sherakama — APPROVED** [review]
Just one pedantic request on where to put the `.get` and then GTG.

**Comment by stfurkan** [comment]
> Just one pedantic request on where to put the `.get` and then GTG.

Thanks for the review @sherakama :) I moved `get` after the `use`. I'll merge the PR after all the tests pass.

**Pull request merged**